react/modules/tasks/routing.md

```diff
@@ -0,0 +1,75 @@
+# React Routing. Tests.
+
+## Technical Requirements.
+
+1. Create a separate branch for this task from the previous task's branch.
+2. All components must be changed to **functional components**, except **Error Boundary** components, as error boundaries in React still need to be class components.
+   - Implement custom hook to restore search query from the local storage (LS) and put it to the LS on unmount. Use respective React lifecycle hook as a basis.
+3. All logic should be split into components:
+   - If you need an access either to the component's lifecycle or the state **use hooks**.
+   - All data should be stored in the **component's state**.
+4. Add routing to your application using **React Router**.
+5. Add 404 page when user navigates to non-existing route.
+6. Add and configure test runner: Jest or Vitest. Test runner should show the test coverage. You should aim to reach at least 80% of the test coverage.
+7. Add testing library: React Testing Library. You should add tests for the several scenarios keeping in mind that mocked data should be used instead of real API calls.
```

Comment:
Add a 404 page
Add and configure a test runner
Add a testing library

Reply:
done